cassandra-diagnostics-reporter-kafka/src/main/java/io/smartcat/cassandra/diagnostics/reporter/KafkaReporter.java

```diff
@@ -49,7 +49,7 @@ public KafkaReporter(ReporterConfiguration configuration) {
         properties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
 
         producer = new KafkaProducer<>(properties);
-        partitionKey = Utils.getHostname();
+        partitionKey = Utils.getSystemname() + "_" + Utils.getHostname();
```

Comment:
systemName and hostname should be fetched from measurement's tags collection. This way, it cannot be overridden by the configuration.